software/nanolyse/main.nf

```diff
@@ -0,0 +1,42 @@
+// Import generic module functions
+include { initOptions; saveFiles; getSoftwareName } from './functions'
+
+params.options = [:]
+def options    = initOptions(params.options)
+
+process NANOLYSE {
+    tag "$meta.id"
+    label 'process_low'
+    publishDir "${params.outdir}",
+        mode: params.publish_dir_mode,
+        saveAs: { filename -> saveFiles(filename:filename, options:params.options, publish_dir:getSoftwareName(task.process), meta:meta, publish_by_meta:['id']) }
+
+    conda     (params.enable_conda ? "bioconda::nanolyse=1.2.0" : null)
+    container "quay.io/biocontainers/nanolyse:1.2.0--py_0"
+
+    input:
+    tuple val(meta), path(fastq)
+    path nanolyse_fasta
+
+    output:
+    tuple val(meta), path("*.fastq.gz")   ,emit: nanolyse_fastq
+    path "*.log"                          ,emit: nanolyse_log
+    path "*.version.txt"                  ,emit: version
+
+    script:
+    """
+    gunzip -c $fastq | NanoLyse -r $nanolyse_fasta | gzip > ${meta.id}.clean.fastq.gz
+    cp NanoLyse.log ${meta.id}.nanolyse.log
+    NanoLyse --version &> nanolyse.version.txt
+    """
+}
+
+process GET_NANOLYSE_FASTA {
+    output:
+    path "*fasta.gz"  , emit: nanolyse_fasta
+
+    script:
+    """
+    wget https://github.com/wdecoster/nanolyse/raw/master/reference/lambda.fasta.gz
+    """
+}
```

Comment:
What is this used for? This should probably not be in the module.

Reply:
Hi @KevinMenden, this process is for staging the `nanolyse` fasta file provided by `nanolyse`. This fasta file is only used by the `NANOLYSE` process, so I'm not sure whether I should separate this `GET_NANOLYSE_FASTA` process into another module.

Reply:
Hi @yuukiiwa I would say you should separate it. 

To me this look rather like something that would be a local module :thinking: not sure whether we need to include it in nf-core/modules.
Although you could make it a submodule of nanlyse.

Reply:
Thanks! I think I will remove it here and add it locally for the test.

Reply:
Hi @KevinMenden, I have moved the `GET_NANOLYSE_FASTA` process to the `NANOLYSE` test as a local module. Thanks!